libcxx/test/std/experimental/simd/simd.mask.class/simd_mask_copy.pass.cpp

```diff
@@ -0,0 +1,139 @@
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14
+
+// <experimental/simd>
+//
+// [simd.mask.class]
+// template<class Flags> void copy_from(const value_type* mem, Flags);
+// template<class Flags> void copy_to(value_type* mem, Flags);
+
+#include "../test_utils.h"
+#include <experimental/simd>
+
+namespace ex = std::experimental::parallelism_v2;
+
+// credit to: https://stackoverflow.com/a/466242
+constexpr size_t next_pow2(size_t v) noexcept {
+  v--;
+  v |= v >> 1;
+  v |= v >> 2;
+  v |= v >> 4;
+  v |= v >> 8;
+  v |= v >> 16;
+  v++;
+  return v;
+}
+
+struct CheckSimdCopyFrom {
+  template <class _Tp, class SimdAbi, size_t _Np>
+  void operator()() {
+    {
+      constexpr auto alignas_size = ex::memory_alignment_v<ex::simd_mask<_Tp, SimdAbi>>;
+      ex::simd_mask<_Tp, SimdAbi> origin_simd_mask{};
+
+      constexpr auto array_length = origin_simd_mask.size();
+
+      alignas(alignas_size) bool buffer[array_length];
+      for (size_t i = 0; i < array_length; i++)
+        buffer[i] = static_cast<bool>(i + 1);
+
+      origin_simd_mask.copy_from(buffer, ex::vector_aligned_tag());
+
+      assert_simd_mask_value_correct(origin_simd_mask, buffer);
+    }
+
+    {
+      constexpr auto alignas_size = std::max(next_pow2(sizeof(_Tp)), next_pow2(_Np));
+      ex::simd_mask<_Tp, SimdAbi> origin_simd_mask{};
+      constexpr auto array_length = origin_simd_mask.size();
+
+      alignas(alignas_size) bool buffer[array_length];
+      for (size_t i = 0; i < array_length; i++)
+        buffer[i] = static_cast<bool>(i + 1);
+
+      origin_simd_mask.copy_from(buffer, ex::overaligned_tag<alignas_size>());
+
+      assert_simd_mask_value_correct(origin_simd_mask, buffer);
+    }
+
+    {
+      constexpr auto alignas_size = std::max(next_pow2(sizeof(_Tp)), next_pow2(alignof(_Tp)));
+      ex::simd_mask<_Tp, SimdAbi> origin_simd_mask{};
+      constexpr auto array_length = origin_simd_mask.size();
+
+      alignas(alignas_size) bool buffer[array_length];
+      for (size_t i = 0; i < array_length; i++)
+        buffer[i] = static_cast<bool>(i + 1);
+
+      origin_simd_mask.copy_from(buffer, ex::element_aligned_tag());
+
+      assert_simd_mask_value_correct(origin_simd_mask, buffer);
+    }
+  }
+};
+
+struct CheckSimdCopyTo {
+  template <class _Tp, class SimdAbi, size_t _Np>
+  void operator()() {
+    {
+      constexpr auto alignas_size = ex::memory_alignment_v<ex::simd_mask<_Tp, SimdAbi>>;
+
+      const ex::simd_mask<_Tp, SimdAbi> origin_simd_mask([](bool i) { return static_cast<bool>(i + 1); });
+      constexpr auto array_length = origin_simd_mask.size();
+      alignas(alignas_size) bool expected_buffer[array_length]{};
+
+      origin_simd_mask.copy_to(expected_buffer, ex::vector_aligned_tag());
+
+      for (size_t i = 0; i < origin_simd_mask.size(); i++)
+        assert(expected_buffer[i] == static_cast<bool>(i + 1));
```

Comment:
Use `assert_simd_mask_value_correct` to check. Please check all tests for such problems. I will not mark them again